tensorboard/backend/event_processing/event_file_loader_test.py

```diff
@@ -19,94 +19,136 @@
 from __future__ import division
 from __future__ import print_function
 
+import abc
+import io
 import os
 import tempfile
 
+import six
 import tensorflow as tf
 
 
 from tensorboard.backend.event_processing import event_file_loader
+from tensorboard.compat.proto import event_pb2
+from tensorboard.summary.writer import record_writer
+
+
+FILENAME = "test.events"
+
+
+@six.add_metaclass(abc.ABCMeta)
+class EventFileLoaderTestBase(object):
+    def _append_record(self, data):
+        with open(os.path.join(self.get_temp_dir(), FILENAME), "ab") as f:
+            record_writer.RecordWriter(f).write(data)
+
+    def _make_loader(self):
+        return self._loader_class(os.path.join(self.get_temp_dir(), FILENAME))
+
+    @abc.abstractproperty
+    def _loader_class(self):
+        """Returns the Loader class under test."""
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def assertEventWallTimes(self, load_result, event_wall_times_in_order):
+        """Asserts that loader.Load() result has specified event wall times."""
+        raise NotImplementedError()
+
+    def testLoad_emptyEventFile(self):
+        with open(os.path.join(self.get_temp_dir(), FILENAME), "ab") as f:
+            f.write(b"")
+        loader = self._make_loader()
+        self.assertEmpty(list(loader.Load()))
+
+    def testLoad_staticEventFile(self):
+        self._append_record(_make_event(wall_time=1.0))
+        self._append_record(_make_event(wall_time=2.0))
+        self._append_record(_make_event(wall_time=3.0))
+        loader = self._make_loader()
+        self.assertEventWallTimes(loader.Load(), [1.0, 2.0, 3.0])
+
+    def testLoad_dynamicEventFile(self):
+        self._append_record(_make_event(wall_time=1.0))
+        loader = self._make_loader()
+        self.assertEventWallTimes(loader.Load(), [1.0])
+        self._append_record(_make_event(wall_time=2.0))
+        self.assertEventWallTimes(loader.Load(), [2.0])
+        self.assertEmpty(list(loader.Load()))
```

Comment:
Huh, I’d always used `assertFalse` for this. [TIL!][absl]

[absl]: https://github.com/abseil/abseil-py/blob/06edd9c20592cec39178b94240b5e86f32e19768/absl/testing/absltest.py#L820-L821